Fix scenario-test workflow

There were some errors in `scenario-test` :

https://github.com/autowarefoundation/autoware/actions/runs/14752704201/job/41413644178?pr=6092#step:16:34

> `WARNING: test_results.log is missing! The simulation might have crashed.`

https://github.com/autowarefoundation/autoware/actions/runs/14752704201/job/41413644178?pr=6092#step:16:50

> `output_directory                    := ./home/vishalchauhan/results`

@vish0012 this path might need to be updated.

https://github.com/autowarefoundation/autoware/actions/runs/14752704201/job/41413644178?pr=6092#step:16:71

> `[scenario_test_runner.py-1] Error: failed validating '1' with XsdPatternFacets(['[$][{][ A-Za-z0-9_\\+\\-\\*/%$\\(\\)\\.,]*[\\}]']):`

_Originally posted by @xmfcx in https://github.com/autowarefoundation/autoware/issues/6092#issuecomment-2841737731_
            

@vish0012 can you accept the invitation to this organization that came to your github e-mail?

It seems like scenario simulator fails locally as well.

## How to test
1. `vcs import src < autoware.repos`
1. `vcs import src < simulator.repos`
1. `colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release  --continue-on-error`
1. Run scenario simulator
```
  ros2 launch scenario_test_runner scenario_test_runner.launch.py \
  architecture_type:=awf/universe/20250130 \
  record:=false \
  scenario:='$(find-pkg-share scenario_test_runner)/scenario/sample.yaml' \
  sensor_model:=sample_sensor_kit \
  vehicle_model:=sample_vehicle
```

It fails with the error saying:
```
[scenario_interpreter_node-3] [ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'tier4_deprecated_api_adapter' not found.
```

## How to solve the issue
It requires two changes.
1. Use the latest version of tier4_ad_api_adaptor, which includes tier4_deprecated_api_adapter. I've already created the release PR for this: https://github.com/tier4/tier4_ad_api_adaptor/pull/150
2. Add patch for dupcliate node checker to autoware_launch. The latest scenario simulator seems to produce many nodes called `getParameter` which causes error in Autoware. Until the issue is solved in the upstream, we need the patch as a workaround.  https://github.com/autowarefoundation/autoware_launch/pull/1438

--- a/.github/workflows/scenario-test.yaml
+++ b/.github/workflows/scenario-test.yaml
@@ -130,17 +130,20 @@ jobs:
         with:
           name: scenario-test-results
           path: ./results
-          
+
       - name: Parse results and capture diagnostics
         run: |
           RESULT_FILE="./results/test_results.log"
           XML_FILE="./results/scenario_test_runner/result.junit.xml"
+
           if [ -f "$RESULT_FILE" ]; then
             echo "Parsing test results..."
             FAILURES=$(grep -i "failure" "$RESULT_FILE" | wc -l)
             ERRORS=$(grep -i "error" "$RESULT_FILE" | wc -l)
+
             echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" > "$XML_FILE"
             echo "<testsuite name='ScenarioTest' failures='$FAILURES' errors='$ERRORS'>" >> "$XML_FILE"
+
             FOUND_ISSUES=0
             while IFS= read -r line; do
               if echo "$line" | grep -qi "failure"; then
@@ -155,8 +158,10 @@ jobs:
                 FOUND_ISSUES=1
               fi
             done < "$RESULT_FILE"
+
             echo "</testsuite>" >> "$XML_FILE"
             echo "Results parsed and saved in $XML_FILE"
+
             if [ $FOUND_ISSUES -eq 1 ]; then
               echo "ERROR: Detected failures or errors in the simulation."
               exit 1

--- a/.github/workflows/scenario-test.yaml
+++ b/.github/workflows/scenario-test.yaml
@@ -7,7 +7,6 @@ on:
     paths:
       - simulator*.repos
       - .github/workflows/scenario-test.yaml
-
 jobs:
   scenario-test:
     runs-on: ubuntu-24.04
@@ -15,10 +14,8 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
-
       - name: Free disk space
         uses: ./.github/actions/free-disk-space
-
       - name: Install required packages
         run: |
           apt-get update
@@ -32,7 +29,6 @@ jobs:
             ros-humble-pcl-ros
           pip install --upgrade gdown vcstool
         shell: bash
-
       - name: Clone Autoware and Import simulator.repos
         run: |
           mkdir -p ~/autoware_ws/src
@@ -41,14 +37,16 @@ jobs:
           cd src
           vcs import < autoware/simulator.repos
         shell: bash
-
       - name: Install ROS dependencies
         run: |
           cd ~/autoware_ws
           rosdep update
           rosdep install --from-paths src --ignore-src -r -y
         shell: bash
-
+      - name: Downgrade xmlschema
+        run: |
+          pip3 install --user xmlschema==3.4.5
+        shell: bash
       - name: Build Autoware
         run: |
           cd ~/autoware_ws
@@ -57,18 +55,15 @@ jobs:
           colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
           source install/setup.bash
         shell: bash
-
       - name: Check Autoware installation
         run: |
           source ~/autoware_ws/install/setup.bash
           ros2 pkg list | grep autoware
         shell: bash
-
       - name: Download scenario file
         run: |
           gdown --id 1Tq7snfDPsuPHPtl50aL5fJY6paiC-HxV -O sample-scenario.yaml
         shell: bash
-
       - name: Download sample-map-planning maps
         # cspell:disable
         run: |
@@ -77,12 +72,10 @@ jobs:
           unzip sample-map-planning.zip -d /home/user/autoware_map
         # cspell:enable
         shell: bash
-
       - name: Fix path in sample-scenario.yaml
         run: |
           sed -i "s@/home/user/sample-map-planning/pointcloud_map.pcd@/home/user/autoware_map/sample-map-planning/pointcloud_map.pcd@g" sample-scenario.yaml
         shell: bash
-
       - name: Build workspace
         run: |
           cd ~/autoware_ws
@@ -90,13 +83,11 @@ jobs:
           colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
           source install/setup.bash
         shell: bash
-
       - name: Check lanelet parameters in config file
         run: |
           echo "Lanelet-related parameters in the configuration:"
           grep -i "lanelet" /github/home/autoware_ws/install/scenario_test_runner/share/scenario_test_runner/config/parameters.yaml || true
         shell: bash
-
       - name: Launch scenario_test_runner with diagnostics
         run: |
           SCENARIO_FILE="./sample-scenario.yaml"
@@ -119,26 +110,21 @@ jobs:
             exit 1
           fi
         shell: bash
-
       - name: Upload artifacts
         uses: actions/upload-artifact@v4
         with:
           name: scenario-test-results
           path: ./results
-
       - name: Parse results and capture diagnostics
         run: |
           RESULT_FILE="./results/test_results.log"
           XML_FILE="./results/scenario_test_runner/result.junit.xml"
-
           if [ -f "$RESULT_FILE" ]; then
             echo "Parsing test results..."
             FAILURES=$(grep -i "failure" "$RESULT_FILE" | wc -l)
             ERRORS=$(grep -i "error" "$RESULT_FILE" | wc -l)
-
             echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" > "$XML_FILE"
             echo "<testsuite name='ScenarioTest' failures='$FAILURES' errors='$ERRORS'>" >> "$XML_FILE"
-
             FOUND_ISSUES=0
             while IFS= read -r line; do
               if echo "$line" | grep -qi "failure"; then
@@ -153,10 +139,8 @@ jobs:
                 FOUND_ISSUES=1
               fi
             done < "$RESULT_FILE"
-
             echo "</testsuite>" >> "$XML_FILE"
             echo "Results parsed and saved in $XML_FILE"
-
             if [ $FOUND_ISSUES -eq 1 ]; then
               echo "ERROR: Detected failures or errors in the simulation."
               exit 1

--- a/.github/workflows/scenario-test.yaml
+++ b/.github/workflows/scenario-test.yaml
@@ -7,6 +7,7 @@ on:
     paths:
       - simulator*.repos
       - .github/workflows/scenario-test.yaml
+
 jobs:
   scenario-test:
     runs-on: ubuntu-24.04
@@ -14,8 +15,10 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
+
       - name: Free disk space
         uses: ./.github/actions/free-disk-space
+
       - name: Install required packages
         run: |
           apt-get update
@@ -29,6 +32,7 @@ jobs:
             ros-humble-pcl-ros
           pip install --upgrade gdown vcstool
         shell: bash
+
       - name: Clone Autoware and Import simulator.repos
         run: |
           mkdir -p ~/autoware_ws/src
@@ -37,16 +41,19 @@ jobs:
           cd src
           vcs import < autoware/simulator.repos
         shell: bash
+
       - name: Install ROS dependencies
         run: |
           cd ~/autoware_ws
           rosdep update
           rosdep install --from-paths src --ignore-src -r -y
         shell: bash
+
       - name: Downgrade xmlschema
         run: |
           pip3 install --user xmlschema==3.4.5
         shell: bash
+
       - name: Build Autoware
         run: |
           cd ~/autoware_ws
@@ -55,15 +62,18 @@ jobs:
           colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
           source install/setup.bash
         shell: bash
+
       - name: Check Autoware installation
         run: |
           source ~/autoware_ws/install/setup.bash
           ros2 pkg list | grep autoware
         shell: bash
+
       - name: Download scenario file
         run: |
           gdown --id 1Tq7snfDPsuPHPtl50aL5fJY6paiC-HxV -O sample-scenario.yaml
         shell: bash
+
       - name: Download sample-map-planning maps
         # cspell:disable
         run: |
@@ -72,10 +82,12 @@ jobs:
           unzip sample-map-planning.zip -d /home/user/autoware_map
         # cspell:enable
         shell: bash
+
       - name: Fix path in sample-scenario.yaml
         run: |
           sed -i "s@/home/user/sample-map-planning/pointcloud_map.pcd@/home/user/autoware_map/sample-map-planning/pointcloud_map.pcd@g" sample-scenario.yaml
         shell: bash
+
       - name: Build workspace
         run: |
           cd ~/autoware_ws
@@ -83,11 +95,13 @@ jobs:
           colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
           source install/setup.bash
         shell: bash
+
       - name: Check lanelet parameters in config file
         run: |
           echo "Lanelet-related parameters in the configuration:"
           grep -i "lanelet" /github/home/autoware_ws/install/scenario_test_runner/share/scenario_test_runner/config/parameters.yaml || true
         shell: bash
+
       - name: Launch scenario_test_runner with diagnostics
         run: |
           SCENARIO_FILE="./sample-scenario.yaml"
@@ -110,11 +124,13 @@ jobs:
             exit 1
           fi
         shell: bash
+
       - name: Upload artifacts
         uses: actions/upload-artifact@v4
         with:
           name: scenario-test-results
           path: ./results
+          
       - name: Parse results and capture diagnostics
         run: |
           RESULT_FILE="./results/test_results.log"

